### ISSUE TITLE
Adventure map, AI makes useless unboarding on a single tile

### ISSUE THREAD

**Issue by Branikolog** [ihhub/fheroes2#9129]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

AI unboards and jumps back on a board on the next turn for no reason.

![image](https://github.com/user-attachments/assets/558a7abd-1f05-4564-97de-6754356d42bf)

https://github.com/user-attachments/assets/1a1e9f8c-49b8-4c33-921a-24b6521d8ef1

[AI unboard fail.zip](https://github.com/user-attachments/files/17097266/AI.unboard.fail.zip)

### Describe the possible solution

AI shouldn't make such useless steps.

**Comment by Branikolog** [comment]
@oleg-derevenetz 
Thanks for clarification!
How do you think, is it possible to teach AI not to waste so much points for just one step in future? Or allow him such movements only if there're no other tasks nearby?
I'm going to leave this issue opened, as such cases heavily slow down AI.

**Comment by oleg-derevenetz** [comment]
> Or allow him such movements only if there're no other tasks nearby?

Fog discovery is already considered as a last resort task :)

**Comment by Branikolog** [comment]
Indeed. But I'm 100% sure has not revealed the whole map, since there're two treasure chests above. I expect him to disembark there, but not in the only tile in forest. :)
I suggest forbidding AI to disembark on such tiles belonging to small closed territories, where he cannot make any action. :)

**Comment by oleg-derevenetz** [comment]
> I suggest forbidding AI to disembark on such tiles belonging to small closed territories, where he cannot make any action. :)

In general case, you may never know for sure. Even if this particular hero cannot do any action, some other hero can, if scout found something valuable enough - e.g. enemy castle or a hero, and that other hero has the Dimension Door spell. I really don't care if AI actions look "strange" from a human player's point of view as long as they are backed by logic.
